test/jdk/java/awt/regtesthelpers/PassFailJFrame.java

```diff
@@ -0,0 +1,156 @@
+import javax.swing.SwingUtilities;
+
+public class PassFailJFrame extends JFrame {
+    private final CountDownLatch latch = new CountDownLatch(1);
+    private boolean failed = false;
+    private String testFailedReason;
+    private JTextArea instructionsText;
+    private final int maxRowLength;
+    private final int maxStringLength;
+    private final int timeoutMinutes;
+
+    /**
+     * Constructs a JFrame with a given title & serves as test instructional
+     * frame where the user follows the specified test instruction in order
+     * to test the test case & mark the test pass or fail. If the expected
+     * result is seen then the user click on the 'Pass' button else click
+     * on the 'Fail' button and the reason for the failure should be
+     * specified in the JDailog JTextArea.
+     *
+     * @param title           title of the Frame.
+     * @param instructions    specified instruction that user should follow.
+     * @param maxRowLength    number of visible rows of the JTextArea where the
+     *                        instruction is show.
+     * @param maxStringLength number of columns of the instructional JTextArea
+     * @param timeoutMinutes  timeout of the test where time is specified in
+     *                        minutes.
+     * @throws HeadlessException
+     * @throws InterruptedException
+     * @throws InvocationTargetException
+     */
+    public PassFailJFrame(String title, String instructions,
+                          int maxRowLength, int maxStringLength,
+                          int timeoutMinutes) throws HeadlessException,
+            InterruptedException, InvocationTargetException {
+        super(title);
+        this.maxRowLength = maxRowLength;
+        this.maxStringLength = maxStringLength;
+        this.timeoutMinutes = timeoutMinutes;
+
+        SwingUtilities.invokeAndWait(() -> {
+            setLayout(new BorderLayout());
+            instructionsText = new JTextArea(instructions, maxRowLength, maxStringLength);
+            instructionsText.setEditable(false);
+            instructionsText.setFocusable(false);
+            add(instructionsText, BorderLayout.NORTH);
+
+            JButton btnPass = new JButton("Pass");
+            btnPass.addActionListener((e) -> latch.countDown());
+
+            JButton btnFail = new JButton("Fail");
+            btnFail.addActionListener((e) -> getFailureReason());
+
+            JPanel buttonsPanel = new JPanel();
+            buttonsPanel.add(btnPass);
+            buttonsPanel.add(btnFail);
+
+            add(buttonsPanel, BorderLayout.SOUTH);
+            pack();
+            setLocation(10, 10);
+            setVisible(true);
+        });
+    }
+
+    /**
+     * Wait for the user decision i,e user selects pass or fail button.
+     * If user does not select pass or fail button then the test waits for
+     * the specified timeoutMinutes period and the test gets timeout.
+     * This method should not be invoke with in EDT.
```

Comment:
This method should not be invoked on the EDT ?

Actually I wonder if we can do better  for this AND the other method.

CHECK if its the EDT and if it is invoke dispose() directly, if it isn't  then use invokeAndWait.

Also remember that SwingUtilities.invokeAndWait is just a wrapper for EventQueue.invokeAndWait

Reply:
Fixed as suggested.